### PR TITLE
ENH : better error message when unit is not known

### LINKED ISSUE #2878: Warning message in set_channel_types causes KeyError
Hi there. I have some data which I read in using this code:

`raw = mne.io.Raw(f, preload=True)`

If I then take a look at an individual channel in the info structure, it looks like this:
```
raw.info['chs'][ch_names.index('EEG 1')]
{'cal': 1.0,
 'ch_name': u'EEG 1',
 'coil_type': 0,
 'coord_frame': 0,
 'kind': 0,
 'loc': array([ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]),
 'logno': 1,
 'range': 1.0,
 'scanno': 1,
 'unit': 0,
 'unit_mul': 0}
```

Now I want to set the channel type to 'misc' for all channels using set_channel_types(), but it blows up:
```
ch_types = dict([(chnm, 'misc') for chnm in raw.ch_names])
raw.set_channel_types(ch_types)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-14-fc7277662040> in <module>()
      1 ch_types = dict([(chnm, 'misc') for chnm in raw.ch_names])
----> 2 raw.set_channel_types(ch_types)

/home/despoB/arjun/anaconda2/lib/python2.7/site-packages/mne/channels/channels.pyc in set_channel_types(self, mapping)
    289                 warnings.warn("The unit for Channel %s has changed "
    290                               "from %s to %s." % (ch_name,
--> 291                                                   _unit2human[unit_old],
    292                                                   _unit2human[unit_new]))
    293             self.info['chs'][c_ind]['unit'] = _human2unit[ch_type]

KeyError: 0

> /home/despoB/arjun/anaconda2/lib/python2.7/site-packages/mne/channels/channels.py(291)set_channel_types()
    290                               "from %s to %s." % (ch_name,
--> 291                                                   _unit2human[unit_old],
    292                                                   _unit2human[unit_new]))
```

It blows up because unit_old was 0, and _unit2human doesn't have a key 0. This warning message should check unit_old exists in _unit2human or something.

### PR DESCRIPTION
**Pull request by agramfort**
closes #2878
closes #2879

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Heh, I was just about to take this over

**Comment by agramfort** [comment]
you're too slow :)

**Comment by Eric89GXL** [comment]
LGTM, will merge once the CIs are happy.

Now stop messing around, take over #2838, decide it's too much work to get right, and merge my "solution" :)

**Comment by agramfort** [comment]
ok boss ...

File: mne/channels/channels.py [hunk 1]
```diff
@@ -286,6 +286,10 @@ def set_channel_types(self, mapping):
             _check_set(self.info['chs'][c_ind], self.info['projs'], ch_type)
             unit_old = self.info['chs'][c_ind]['unit']
             unit_new = _human2unit[ch_type]
+            if unit_old not in _unit2human:
+                raise ValueError("Channel '%s' has unknown unit (%s). Please fix"
```
Comment (Eric89GXL) [inline mne/channels/channels.py]: line too long :(

**Comment by Eric89GXL** [comment]
(I'll push to your branch)

**Comment by agramfort** [comment]
thx 

merge when green

**Pull request merged**